docs/source/user-guide/tasks/work-with-packages.rst

```diff
@@ -13,19 +44,136 @@ environment:
+
+  anaconda-project add-packages --pip package1 package2
+  anaconda-project remove-packages --pip package1 package2
+
+NOTE: Replace ``package1`` and ``package2`` with the names of
+the packages that you want to include. You can specify as many
+packages as you want.
+
+EXAMPLE: To add the ``requests`` package to the default environment::
+
+  $ anaconda-project add-packages --pip requests
+  Collecting requests
+    Using cached requests-2.25.1-py2.py3-none-any.whl (61 kB)
+  Requirement already satisfied: certifi>=2017.4.17 in ./envs/default/lib/python3.8/site-packages (from requests) (2020.12.5)
+  Collecting idna<3,>=2.5
+    Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
+  Collecting chardet<5,>=3.0.2
+    Using cached chardet-4.0.0-py2.py3-none-any.whl (178 kB)
+  Collecting urllib3<1.27,>=1.21.1
+    Using cached urllib3-1.26.4-py2.py3-none-any.whl (153 kB)
+  Installing collected packages: urllib3, idna, chardet, requests
+  Successfully installed chardet-4.0.0 idna-2.10 requests-2.25.1 urllib3-1.26.4
+  Using Conda environment /Users/adefusco/Desktop/testproj/envs/default.
+  Added packages to project file: requests.
+
+Optionally, you can edit the ``anaconda-project.yml`` file to add packages using
+the ``pip:`` key within the ``packages:`` list. For example,
+
+.. code-block:: yaml
+
+  name: ExampleProject
+
+  packages:
+    - python=3.8
+    - notebook
+    - pandas
+    - pip:
+      - requests
+  
+  env_specs:
+    default: {}
+
+Then run ``anaconda-project prepare`` to install the new packages into the environment.
+
+EXAMPLE: To remove the ``requests`` package from the default environment::
+
+  $ anaconda-project remove-packages --pip requests
+  Found existing installation: requests 2.25.1
+  Uninstalling requests-2.25.1:
+    Successfully uninstalled requests-2.25.1
+  Using Conda environment /Users/adefusco/Desktop/testproj/envs/default.
+  Removed packages from project file: requests.
+
+
+Pip package specifications
+==========================
+
+Pip packages can specified in a number of ways.
+
+* From PyPI (or other indexes)
+* Direct URL to the package archive
+* Revision Control services (for example git and svn)
+* Package directories inside the Anaconda Project itself
+
+To install a package from a revision control service::
+
+  anaconda-project add-packages --pip git+<protocol>://<revision-control-domain>/<repository.git>[version-branch]#egg=<package-name>
+
+Where
+
+* ``<protocol>`` is the web protocol of the domain: i.e, ``http`` or ``https``
+* ``<revision-control-domain>`` is the URL of the service: i.e. ``github.com``
+* ``<repository.git>`` is the name of the revision control repository, you can include the branch name or release tag here.
+* ``[version-branch]`` optionally install a specific version or branch of the repository
+* ``<package>`` is the name of the package as declared in ``setup.py``
+
+NOTE: It is required that you use ``#egg=<package>`` to install a revision control hosted
+package.
```

Comment:
I think it's worth explaining _why_ you need to specify `egg`. I believe you; I'm pretty sure that there are important cases where it won't work without that. But I'm also pretty sure that I recall there are important cases where it does work without `egg`, and so it's helpful to specify here what those are. E.g. maybe it worked without `egg` from the command line with pip, but not in a-p?  I can't remember, but it's helpful to explain what would go wrong so that people who forget `egg` can detect that and recover from it.

Also, can you clarify whether this syntax is to do with a-p, or just pip? As far as I know this is pip syntax, in which case it would be useful to point to the relevant pip documentation for the definitive description of this functionality.

Reply:
I've update the note

```
NOTE: It is required that you use ``#egg=<package>`` to install a revision control hosted
package. This is considered best practice by pip and allows the pip dependency solver to 
correctly identify the package if it is a dependent of another package in your project.
```

See https://pip.pypa.io/en/latest/cli/pip_install/#vcs-support